app/models/child_benefit_tax_calculator.rb

```diff
@@ -83,28 +83,45 @@ def selected_tax_year
     TAX_YEARS[@tax_year.to_s]
   end
 
-  def can_estimate?
-    @total_annual_income.positive? && can_calculate?
-  end
-
   def benefits_claimed_amount
-    all_weeks_children = {}
-    full_year_children = @children_count - @part_year_children_count
-    (child_benefit_start_date...child_benefit_end_date).each_slice(7) do |week|
-      monday = monday_on_or_after(week.first)
-      all_weeks_children[monday] = 0
-      @starting_children.each do |child|
-        all_weeks_children[monday] += 1 if eligible?(child, tax_year, monday)
-      end
-      full_year_children.times do
-        all_weeks_children[monday] += 1
-      end
+    no_of_full_year_children = @children_count - @part_year_children_count
+    first_child_calculated = false
+    total_benefit_amount = 0
+
+    if no_of_full_year_children.positive?
+      no_of_weeks = total_number_of_mondays(child_benefit_start_date, child_benefit_end_date)
+      no_of_additional_children = no_of_full_year_children - 1
+      total_benefit_amount = first_child_rate_total(no_of_weeks) + additional_child_rate_total(no_of_weeks, no_of_additional_children)
+      first_child_calculated = true
+    else
+      first_child_calculated = false
     end
-    # calculate total for all weeks
-    total = all_weeks_children.values.inject(0) do |sum, n|
-      sum + BigDecimal(weekly_sum_for_children(n).to_s)
+
+    if @starting_children.count.positive?
+      first_child = 0
+
+      @starting_children.each_with_index do |child, index|
+        start_date = if (child.start_date < child_benefit_start_date) || ((@tax_year == 2012) && (child.start_date < TAX_COMMENCEMENT_DATE))
+                       child_benefit_start_date
+                     else
+                       child.start_date
+                     end
+
+        end_date = if child.end_date.nil? || (child.end_date > child_benefit_end_date)
+                     child_benefit_end_date
+                   else
+                     child.end_date
+                   end
+
+        no_of_weeks = total_number_of_mondays(start_date, end_date)
+        total_benefit_amount = if index.equal?(first_child) && (first_child_calculated == false)
+                                 total_benefit_amount + first_child_rate_total(no_of_weeks)
+                               else
+                                 total_benefit_amount + additional_child_rate_total(no_of_weeks, 1)
```

Comment:
Ah this is a workaround so that `additional_child_rate_total` can be reused here.
When we are looping through each child and hit `additional_child_rate_total` this is calculation is only applied to the current child (if it's not the first child).

Reply:
Ah okay, yeah that makes sense :+1: